src/main/java/io/zeebe/monitor/zeebe/kafka/importers/KafkaMessageSubscriptionImporter.java

```diff
@@ -0,0 +1,77 @@
+package io.zeebe.monitor.zeebe.kafka.importers;
+
+import io.camunda.zeebe.protocol.record.intent.MessageStartEventSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
+import io.zeebe.monitor.entity.MessageSubscriptionEntity;
+import io.zeebe.monitor.repository.MessageSubscriptionRepository;
+import io.zeebe.monitor.zeebe.kafka.GenericRecord;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.UUID;
+
+@Component
+public class KafkaMessageSubscriptionImporter {
+
+  @Autowired private MessageSubscriptionRepository messageSubscriptionRepository;
+
+  public void importMessageSubscription(final GenericRecord record) {
+
+    final MessageSubscriptionIntent intent =
+        MessageSubscriptionIntent.valueOf(record.getIntent());
+    final long timestamp = record.getTimestamp();
+    Map values = record.getValue();
+    final MessageSubscriptionEntity entity =
+        messageSubscriptionRepository
+            .findByElementInstanceKeyAndMessageName(
+                    values.get("elementInstanceKey") != null ? (Long) values.get("elementInstanceKey") : 0, (String)values.get("messageName"))
```

Comment:
you're basically searching by `elementInstanceKey` `0`

Reply:
if value is then null yes